libcxx/include/__bit/countl.h

```diff
@@ -25,15 +25,27 @@ _LIBCPP_PUSH_MACROS
 _LIBCPP_BEGIN_NAMESPACE_STD
 
 _LIBCPP_NODISCARD inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR int __libcpp_clz(unsigned __x) _NOEXCEPT {
+#if __has_builtin(__builtin_clzg)
+  return __builtin_clzg(__x);
+#else
   return __builtin_clz(__x);
+#endif
```

Comment:
I don't see a real win here. If we can use `__builtin_clzg` unconditionally in a few years we might be able to do more refactoring to remove several of these overloads.

Reply:
Yes, I agree. I also was not very convinced by using the generic variant by any means. Removed the `#ifs` in the abstraction helpers.